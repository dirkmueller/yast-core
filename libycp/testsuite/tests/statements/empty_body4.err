Parsed:
----------------------------------------------------------------------
YError
----------------------------------------------------------------------
[Interpreter] tests/statements/empty_body4.ycp:8 Empty function body
Parsed:
----------------------------------------------------------------------
"we are ok"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
YError
----------------------------------------------------------------------
[Interpreter] tests/statements/empty_body4.ycp:19 Empty function body
