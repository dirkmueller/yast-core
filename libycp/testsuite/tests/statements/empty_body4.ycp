// function body may not be empty
//
// There's something broken with the error handling.

// in this case, the parser sees the milestone and only then reports
// a definition error for func. weird enough
{
  define void func (string a) ``{
  }

  y2milestone ("Hello1");
//  func ("foo");
}
("we are ok")
// but in this case, it sees the milestone, reports the error
// and then tries to attach the parameters to the call -> SEGV
// Note that with no parameters or an integer parameter, it was ok.
{
  define void func (string a) ``{
  }

  y2milestone ("Hello2");
  func ("foo");
}
