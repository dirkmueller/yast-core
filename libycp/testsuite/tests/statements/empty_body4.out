(nil)
("we are ok")
(nil)
